src/features/ajax/instrument/index.js

```diff
@@ -22,35 +22,34 @@ var handlers = ['load', 'error', 'abort', 'timeout']
 var handlersLen = handlers.length
 
 var origRequest = originals.REQ
-var origXHR = globalScope?.XMLHttpRequest
+var origXHR = globalScope.XMLHttpRequest
 
 export class Instrument extends InstrumentBase {
   static featureName = FEATURE_NAME
   constructor(agentIdentifier, aggregator, auto = true) {
     super(agentIdentifier, aggregator, FEATURE_NAME, auto)
-    const agentRuntime = getRuntime(this.agentIdentifier);
 
-    // Don't instrument Chrome for iOS, it is buggy and acts like there are URL verification issues
-    if (!agentRuntime.xhrWrappable) return
+    // Very unlikely, but in case the existing XMLHttpRequest.prototype object on the page couldn't be wrapped.
+    if (!getRuntime(agentIdentifier).xhrWrappable) return
 
-    agentRuntime.features.xhr = true;   // declare that we are using xhr instrumentation
-
-    this.dt = new DT(this.agentIdentifier)
+    this.dt = new DT(agentIdentifier)
 
     this.handler = (type, args, ctx, group) => handle(type, args, ctx, group, this.ee)
-    this.wrappedFetch = getWrappedFetch(this.ee)
+    wrapFetch(this.ee)
     wrapXhr(this.ee)
-    subscribeToEvents(this.agentIdentifier, this.ee, this.handler, this.dt)
+    subscribeToEvents(agentIdentifier, this.ee, this.handler, this.dt)
 
-    this.importAggregator()
+    this.abortHandler = this.#abort;
+    this.importAggregator();
   }
-}
-
-// TODO update all of this to go into class and use this.ee for ee, handle, and register
 
-export function getWrappedFetch(ee, handler) {
-  var wrappedFetch = wrapFetch(ee)
-  return wrappedFetch
+  /** Restoration and resource release tasks to be done if Ajax loader is being aborted. */
+  #abort() {
+    // (Much of this module affects specific XHR instances.)
+    unwrapFetch(this.ee);
+    unwrapXhr(this.ee);
+    this.abortHandler = null; // weakly allow this abort op to run only once
```

Comment:
If something does call `this.abortHandler` after being set to null it'll throw an error

```suggestion
    this.abortHandler = ( ) => {}; // weakly allow this abort op to run only once
```

Reply:
- You can see that it's also replaces a flag in JS error which would return `true` even for empty functions when it should return `false`.
- The existing places that calls it, calls it with optional chaining, so it's safe!